src/main/interface/cli.c

```diff
@@ -445,12 +445,23 @@ static void printValuePointer(const clivalue_t *var, const void *valuePointer, b
 
         switch (var->type & VALUE_MODE_MASK) {
         case MODE_DIRECT:
-            if ((value < var->config.minmax.min) || (value > var->config.minmax.max)) {
-                cliPrintCorruptMessage(value);
+            if ((var->type & VALUE_TYPE_MASK) == VAR_UINT32) {
+                if ((value >> var->config.bitcount) > 0) {
```

Comment:
`value` is `int`, so this won't work for 'bitcount==32' (and is implementation dependent). Also bit shift of register size is implementation dependent (shift may be taken modulo 32)